packages/solid-plotly.js/src/factory.tsx

```diff
@@ -55,221 +55,98 @@ const updateEvents: PlotlyHTMLElementEventName[] = [
+        return {
+          data: el.data as Partial<PlotData>[],
+          layout: el.layout ?? {},
+          frames: el._transitionData?._frames,
         }
-      })
-    }
-
-    const addEventHandler = (eventName: PlotlyHTMLElementEventName, handler: EventHandler) => {
-      const el = plotEl()
-      handlers[eventName] = handler
-      if (isBasicEvent(eventName)) {
-        el?.on(eventName, () => handler(getCurrentFigure(), el))
       }
-    }
 
-    const removeEventHandler = (eventName: PlotlyHTMLElementEventName) => {
-      const el = plotEl()
-      const handler = handlers[eventName]
-      if (handler && isBasicEvent(eventName)) {
-        el?.removeListener(eventName, () => handler(getCurrentFigure(), el))
-        delete handlers[eventName]
-      }
-    }
-
-    // Type guard to check if an event is a basic event
-    const isBasicEvent = (eventName: PlotlyHTMLElementEventName): eventName is PlotlyBasicEvent => {
-      return updateEvents.includes(eventName)
-    }
-
-    const getCurrentFigure = (): PlotlyFigure => {
-      const el = plotEl()
-      return {
-        data: el?.data as Partial<PlotData>[],
-        layout: el?.layout ?? {},
-        frames: el?._transitionData?._frames,
-      }
-    }
-
-    const updatePlotly = async (
-      shouldInvokeResizeHandler: boolean,
-      callback?: EventHandler,
-      shouldAttachUpdateEvents: boolean = false,
-    ) => {
-      try {
-        const el = plotEl()
-        if (!el) {
-          throw new Error('Missing element reference')
-        }
-
-        await Plotly.react(el, props.data, props.layout || {}, props.config)
-
-        syncWindowResize(shouldInvokeResizeHandler)
-        syncEventHandlers()
-        figureCallback(callback)
-
-        if (shouldAttachUpdateEvents) {
-          attachUpdateEvents()
-        }
-      } catch (err) {
-        if (props.onError && err instanceof Error) {
-          props.onError(err)
-        } else {
-          // eslint-disable-next-line no-console
-          console.error('PlotlyComponent error:', err)
+      const updatePlotly = async (callback?: EventHandler) => {
+        try {
+          await Plotly.react(el, props.data, props.layout || {}, props.config)
+          callback?.(getCurrentFigure(), el)
+        } catch (err) {
+          if (props.onError && err instanceof Error) {
+            props.onError(err)
+          } else {
+            // eslint-disable-next-line no-console
+            console.error('PlotlyComponent error:', err)
+          }
         }
       }
-    }
 
-    const retryUpdatePlotly = () => {
-      const el = plotEl()
-      if (el) {
-        updatePlotly(true, props.onInitialized, true)
-      } else {
-        setTimeout(retryUpdatePlotly, 0) // Retry after 1 tick
-      }
+      onMount(() => {
+        initEventHandlers()
+        initUpdateEvents()
+        updatePlotly(props.onInitialized)
+        createEffect(() => updatePlotly(props.onUpdate), { defer: true })
```

Comment:
I don't think the `{ defer: true }` does anything now. It would be the initial value of the parameter or effect, but the value is not used in the effect.

Reply:
a true! good call. 

could do something like

```ts
let initialized = false
createEffect(() => {
  updatePlotly(!initialized ? props.onInitialized : props.onUpdate)
  initialized = true
})
```

then it will call `props.onInitialized` in the initial run and `props.onUpdate` in all the consecutive times the effect is ran

Reply:
pushed it

Reply:
Is it necessary to do any initialization stuff in the `createEffect`? 

Since this is called in `onMount` it will execute after the initial render and `updatePlotly(props.onInitialized)` is called before `createEffect(() => updatePlotly(props.onUpdate))` which setup up the on necessary tracking call. 

Did I understand that correctly?

Reply:
sorry i am not completely following, could you elaborate?

Reply:
All this executes in an `onMount` function. 

```
onMount(() => {
    initEventHandlers()
    initUpdateEvents()

    let initialized = false
    createEffect(() => {
      updatePlotly(!initialized ? props.onInitialized : props.onUpdate)
      initialized = true
    })

    createEffect(() => {
      if (isBrowser && props.useResizeHandler) {
        const resizeHandler = () => Plotly.Plots.resize(el)
        window.addEventListener('resize', resizeHandler)
        onCleanup(() => window.removeEventListener('resize', resizeHandler))
      }
    })

    onCleanup(() => {
      props.onPurge?.()
      Plotly.purge(el)
    })
  })
}
```

So, doing it his way (below) seems fine to me since it's called after the `onMount` and this `updatePlotly(props.onInitialized)` only needs to be call once on mount.

```
updatePlotly(props.onInitialized)
createEffect(() => {  
  updatePlotly(props.onUpdate)
})
```